app/controllers/repositories_controller.rb

```diff
@@ -100,14 +103,21 @@ def branches
   end
 
   def notify_push
```

Comment:
Maybe we should move this into a new controller and split it into before action hooks. What do you think? This way we keep Repositories Controller cohesion.